metaflow/plugins/kfp/kfp.py

```diff
@@ -193,16 +194,51 @@ def __init__(
         self.sqs_role_arn_on_error = sqs_role_arn_on_error
         self._client = None
 
-    def deploy(self, kubernetes_namespace: str, name: Optional[str]) -> Dict[str, Any]:
+    def deploy(
+        self,
+        kubernetes_namespace: str,
+        name: Optional[str],
+        flow_parameters: Dict,
+        recurring_run_enable: Optional[bool] = None,
+        recurring_run_cron: Optional[str] = None,
+        recurring_run_policy: Optional[str] = None,
+        max_run_concurrency: Optional[int] = 10,
+    ) -> Tuple[Dict[str, Any], Dict[str, Any], Dict[str, Any]]:
         try:
             # Register workflow template.
-            workflow: Dict[Text, Any] = self._create_workflow_yaml(
-                output_format="argo-workflow-template", name=name
+            workflow_template: Dict[str, Any] = self._create_workflow_yaml(
+                flow_parameters=flow_parameters,
+                kind="WorkflowTemplate",
+                name=name,
+            )
+            config_map: Dict[str, Any] = KubeflowPipelines._config_map(
+                sanitize_k8s_name(self.name), max_run_concurrency
+            )
+            argo_workflow_name = workflow_template["metadata"]["name"]
+
+            # Create the Argo synchronization ConfigMap
+            config = ArgoClient(
+                namespace=kubernetes_namespace
+            ).create_workflow_config_map(argo_workflow_name, config_map)
+
+            # Create the Argo Workflow Template
+            workflow = ArgoClient(
+                namespace=kubernetes_namespace
+            ).register_workflow_template(argo_workflow_name, workflow_template)
+
+            # Create CronWorkflow
+            cron_workflow: Dict[str, Any] = KubeflowPipelines._cron_workflow(
+                sanitize_k8s_name(self.name),
+                schedule=recurring_run_cron,
+                concurrency=recurring_run_policy,
+                recurring_run_enable=recurring_run_enable,
             )
 
-            return ArgoClient(
+            cron_workflow = ArgoClient(
```

Comment:
I assume this cron_workflow is no longer of `Dict[str, Any]` type. Could you please verify? If it's not maybe type hints need to be updated and this `cron_workflow` should be slightly renamed. Though I can be wrong.